my-app/src/app/courses-page/course-search/course-search.component.spec.ts

```diff
@@ -22,4 +23,20 @@ describe('CourseSearchComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('onClick method should be called', () => {
+    spyOn(component, 'onClick');
+    buttonElem = fixture.debugElement.query(By.css('.search-button'));
+    fixture.detectChanges();
+    buttonElem.triggerEventHandler('click', null);
+    expect(component.onClick).toHaveBeenCalled();
+  });
+
+  it('should show correct result when searc is active', () => {
+    console.log = jasmine.createSpy('log');
+    component.inputValue = searchValue;
+    fixture.detectChanges();
+    component.onClick();
+    expect(console.log).toHaveBeenCalledWith(`You try to find: ${searchValue}`);
```

Comment:
when you update the logic this test is gonna break. Test the behavior. Check if the function is called with a specific argument. 
I think you will have an event emitter in this component. which will emit search text to the parent component. You could test that.